Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs

```diff
@@ -145,8 +145,109 @@ public static FamilyInstance ToRevitFamilyInstance(this Column framingElement, D
         [Output("instance", "Revit FamilyInstance resulting from converting the input BH.oM.Physical.Elements.Pile.")]
         public static FamilyInstance ToRevitFamilyInstance(this Pile framingElement, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
         {
-            BH.Engine.Base.Compute.RecordError(string.Format("Push of pile foundations is not supported in current version of BHoM. BHoM element Guid: {0}", framingElement.BHoM_Guid));
-            return null;
+            if (framingElement == null || document == null)
+                return null;
+
+            FamilyInstance familyInstance = refObjects.GetValue<FamilyInstance>(document, framingElement.BHoM_Guid);
+            if (familyInstance != null)
+                return familyInstance;
+
+            settings = settings.DefaultIfNull();
+
+            BH.oM.Geometry.Line locationLine = framingElement.PileLine();
+            if (locationLine == null)
+                return null;
+
+            Level level = document.LevelBelow(locationLine, settings);
+            if (level == null)
```

Comment:
Hmm I would rather look for closest level above (with potential warning) rather than immediately returning `null`